biscuit-pulsar/src/main/java/com/clevercloud/biscuitpulsar/AuthenticationProviderBiscuit.java

```diff
@@ -60,11 +77,16 @@ public String getAuthMethodName() {
   }
 
   public String authenticate(AuthenticationDataSource authData) throws AuthenticationException {
-    String biscuit = getBiscuit(authData);
-    return parseBiscuit(biscuit);
+    String bearer = getBearerValue(authData);
+
+    if (isJWT(bearer)) {
+      return jwtAuthenticator.authenticate(authData);
+    } else {
+      return parseBiscuit(bearer);
+    }
```

Comment:
@Geal I'm trying isJWT but it's probably better to handle biscuit parser by default and catch exception to try jwt parser then. In fact the JWT detection is that simple and finally become to check the decode content etc. So maybe try catch decode for both biscuit/jwt is better finally. WDYT?

WDYT?

Reply:
yup instead of isJWT (which looks a bit brittle), it would be better to try decoding one way, and if it fails decode the other way